src/corgi-tag-watcher.js

```diff
@@ -0,0 +1,82 @@
+import fetch from 'node-fetch'
+import sax from 'sax'
+import { ensureEnv } from './utils/ensureEnv.js'
+
+export default (app) => {
+  app.on('create', async (context) => {
+    const CORGI_URL = `https://${ensureEnv('CORGI_HOSTNAME')}/api/jobs/`
+    const SLACK_URL = `https://hooks.slack.com/services/${ensureEnv('CORGI_SLACK_SECRET')}`
+    // NOTE: if we miss webhooks look into persistence
+    const logger = app.log.child({ name: 'corgi-tag-watcher' })
+
+    logger.info('recieved webhook.')
+
+    // Do we need to filter events?
+    if (context.payload.ref_type !== 'tag') { return }
+    // get books in repo
+    const contentRequest = context.repo({ path: 'META-INF/books.xml' })
+    const contentMetadata = await context.octokit.repos.getContent(contentRequest)
+    const content = Buffer.from(contentMetadata.data.content, contentMetadata.data.encoding)
+
+    const repo = context.payload.repository.name
+
+    const books = []
+    const parser = sax.parser()
+    parser.onopentag = (node) => {
+      if (node.name === 'BOOK') {
+        books.push(node.attributes.SLUG)
+      }
+    }
+    parser.write(content).close()
+
+    // send job details to CORGI API
+    // make request per book + job type
+    let jobStatus = 'failed to queue'
+
+    try {
+      for (const slug of books) {
+        for (const jobType of [3, 4]) {
+          // 3: git-pdf
+          // 4: git-distribution-preview
```

Comment:
Just to confirm my own understanding (mostly as it may impact load spikes on Concourse since that happened to come up today 😅 ), this will basically queue 6 jobs on a single tag for a bundle with three books, correct?

Reply:
That is correct! :)